### ISSUE TITLE
Flow through data component cannot be instantiated

### ISSUE THREAD

**Issue by lwrage** [osate/osate2#643]
The end to end flow in the following model is not instantiated.

```
package issue643
public
    system top

    end top;

    system implementation top.i
        subcomponents
            p: system S;
            c: system S;
            d: data;
        connections
            c0: data access p.a <-> d;
            c1: data access d <-> c.a;
        flows
            e2e: end to end flow p -> c0 -> d -> c1 -> c;
    end top.i;

    system S
        features
            a: requires data access;
    end S;

end issue643;

```

**Comment by lwrage** [comment]
Fixing #246 will likely help. In general, the flow instantiation must be changed to work for bidirectional connection instances.

**Comment by lwrage** [comment]
It is not clear if we should even have bidirectional connection instance.

**Comment by AaronGreenhouse** [comment]
Fixed by Issue #676

**Comment by AaronGreenhouse** [comment]
Added JUnit tests for this.
